BUM: Improved user creation

AlexVelezLl opened the pull request:
## Summary




* Extracts users table to be its own component.
* Adds date_joined__gt and date_joined__lt filters in the FacilityUserViewset.
* Adds new users page.
* Re-estructure users-related components to be all inside a `users` folder to have a better file organization.
* Moves create user form to be a side panel.
* Moves create user logic from vuex to the side panel itself.
* Adds enroll to a class multi select field.
* Allows enrolling users to classes on user create.
* Prevents unnecessary users reload when naviating to a new page.
* Fixes `identifier` info icon tooltip in users table.

### Screencast

https://github.com/user-attachments/assets/f2c7d222-7dc6-4427-be6b-dd6bc8e49520



## References

Closes #13410.

## Reviewer guidance

* Go to facilities > users.
* Click on add user button.
* Add users using both `save and add another` and `save and continue` actions
* Check that users have been created correctly with their roles and classes.

AlexVelezLl committed:
Move UserPage to users/UsersRootPage

AlexVelezLl committed:
Add base new users page

AlexVelezLl committed:
Move create user page to be a modal

AlexVelezLl committed:
Move side panels to new users folder

AlexVelezLl committed:
Add user create side panel

AlexVelezLl committed:
Extract users table to its own component

AlexVelezLl committed:
Miscellaneous fixes

AlexVelezLl committed:
Use new users table in new users page

AlexVelezLl committed:
Add user bulk actions to new users table

AlexVelezLl committed:
Add enroll in classes

AlexVelezLl committed:
Add new users empty state

AlexVelezLl commented:
@coderabbitai full review

AlexVelezLl commented on kolibri/plugins/facility/assets/src/views/users/sidePanels/UserCreate/index.vue:
```diff
@@ -0,0 +1,481 @@
+        }
+      };
+
+      const submitForm = async () => {
+        formSubmitted.value = true;
+        if (!showPasswordInput.value && !passwordValid.value) {
+          passwordValid.value = true;
+        }
+        if (!formIsValid.value) {
+          return focusOnInvalidField();
+        }
+        busy.value = true;
```
This is the desired behavior.

nucleogenesis reviewed (commented):
I've given the code a solid first pass and nothing jumped out at me. I've also tested locally and things worked as expected. Great work @AlexVelezLl -- on to full QA!

nucleogenesis commented:
@radinamatic 

An additional note of something to focus on, particularly with a11y in mind:

_Multiple-selection classes dropdown_

At the bottom of the creation form, you'll see a dropdown for assigning to classes which opens to present multiple options. I've tested this w/ keyboard nav successfully, but it'd be a great place for your focus.

_No need to focus on much outside the workflow itself for adding multiple users. See the notes below about existing known issues_

__


@AlexVelezLl I found two items of note after I submitted my review (cc @radinamatic so you're aware)

1. I just tested this after I submitted my review :sweat_smile: - can you put the "Are you sure?" modal when trying to close the side panel when the user could lose their work in the form please.
2. There is no value present in the "Creation date" column (although the search works)

radinamatic commented:
On the first cursory QA pass while testing in both Firefox and Chrome on Ubuntu, all workflows seem to be working as specified! 👍🏽 

Two questions/comments:

1. Found mildly confusing that the button ``Save and close`` did nothing if the form was empty. Yes, there was an error notification in the first field, but from the user perspective it seems like the button is not working. I would suggest to check that all form fields are empty (that is a check that form is not partially filled), and in that case pressing the ``Save and close`` should simply close the form, same as the `Close` ✖️ button in the upper right corner. cc @jtamiace 
2. How long do the newly added users remain in the `New users` page? I created several users in one browser, than opened a different browser to repeat the process and previously created users were still there. If that is a _transitory_ container for the new users, seeing there previously created users from another _session_ (different browsers might mean different admins adding users, for example), is also confusing... 🤔 


https://github.com/user-attachments/assets/64d0d35b-b8f5-4363-97ed-45e4c060478b

AlexVelezLl committed:
Show user creation date

AlexVelezLl committed:
Add CloseConfirmationGuard component

nucleogenesis commented:
Thanks for the review @radinamatic - for your note (1) we could change the text on that button from "Close" to "Save & Close" when the form has been changed (and back when it's an empty form)? We won't block this PR on it but we can follow-up on this workflow.

As for (2) it's just a page listing the users added in the last 30 days. Perhaps text explaining that to the user here could help?

pcenov commented:
Hi @AlexVelezLl some additional observations:
1. At `Facility > Classes > Class` we have 2 options - to _assign_ coaches to the class or/and to _enroll_ learners. But when I am creating a new user at the bottom of the form there is a drop-down called `"Enroll in class"` with a checkbox called `"Assign to all classes"`.
This can be confusing because when I was creating a Coach user, I was expecting to actually assign that coach to the class while the coach got enrolled as a learner. So the question is - are we going to support assigning coaches to a class through this form? If not, then it's probably just a matter of changing the text `"Assign to all classes"` to `"Enroll in all classes"`.

https://github.com/user-attachments/assets/5184da36-7716-4099-9f96-b6a93330f6df

2. In your video recording I can see the user type badges displayed next to the name of the non-learner users (Super admin, Admin, Coach and Facility coach) but when I create a new non-learner user they are shown in the table without any user type badges:

![2025-07-10_11-15-31](https://github.com/user-attachments/assets/ac2d278f-8fcb-4be2-b0a2-c0a1c7f10c82)

3. The classes in "Enroll in a class" are not sorted in any way which makes it difficult to find the appropriate class:

![2025-07-10_11-33-15](https://github.com/user-attachments/assets/9bee1d08-e346-4683-9e00-7b98b24f87b0)

4. If I have selected multiple classes then the list should get truncated otherwise it goes out of the field:

![2025-07-10_11-39-20](https://github.com/user-attachments/assets/f8e69b15-d2df-426a-9fbd-a415cea7f032)

5. If I try to enroll a user to an already deleted class I am getting an unhandled error:

https://github.com/user-attachments/assets/2c9f790f-a6f1-4903-883e-f7045dc14d96

6. In mobile view the buttons are centered and should be discussed whether that's the expected layout:

![2025-07-10_11-50-44](https://github.com/user-attachments/assets/629d829e-458f-4de0-9039-aa4ee556b258)

AlexVelezLl committed:
Fix user row index

AlexVelezLl committed:
Switch between enroll to a class and assign to a class when creating a learner or non learner user

AlexVelezLl commented:
Thanks @pcenov!

1. I have fixed this by changing the label and the action when a learner is selected vs when its not. If we are creating a learner it will display enroll to a class and will enroll the learner on save, and when creating a coah or admin the label will say assign to a class and will assign the coach to the class.
2. Just fixed this!
3. I think this can be a follow up issue. I tried to do this, but it seems some other tweaks will be needed.
4. I already opened an issue to solve this in KDS: https://github.com/learningequality/kolibri-design-system/issues/1066. Once we fix this there, it will automatically be fixed here.
5. This can also be in a follow up issue, since we need to make some decissions around how we handle these kind of errors.
6. I have aligned them to the right for now, but would like to hear @tomiwaoLE's or @jtamiace's thoughts.

<img width="469" height="783" alt="image" src="https://github.com/user-attachments/assets/64a68818-445f-47c5-9d3d-8903d9e0261e" />

nucleogenesis reviewed (approved):
Code changes all look good to me - I tested things out locally and the issues @pcenov  mentioned appear to be resolved :+1:

pcenov reviewed (approved):
Excellent @AlexVelezLl - I have filed 2 follow-up issues as suggested and confirm that the ones that you mention as fixed are indeed fixed now.

nucleogenesis commented:
Just noting I smoke tested after the rebase and confirmed that bulk enrollment still works as expected along w/ user creation and it was really nice doing both at once right now :) 

Good to merge once the checks are all green
